src/ensembl/src/content/app/browser/drawer/drawer-views/track-details/TrackDetails.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { getActiveTrackDetails } from 'src/content/app/browser/drawer/drawerSelectors';
+import { fetchTrackDetails } from 'src/content/app/browser/drawer/drawerActions';
+
+import ExternalLink from 'src/shared/components/external-link/ExternalLink';
+import styles from './TrackDetails.scss';
+
+const TrackDetails = () => {
+  const trackDetails = useSelector(getActiveTrackDetails);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!trackDetails) {
+      dispatch(fetchTrackDetails());
```

Comment:
What's the benefit, at this point, to getting track details via redux rather than directly from the component? These details don't seem to be reused in other components; the code that "fetches" them is restricted to the drawer; so why go through redux at all?

_(I understand that in the future, when we start fetching track data for real from a track api, track data will likely be reused by different components, e.g. TrackPanel and Drawer. But if that happens, fetching track data will not be the drawer's concern anyway, so the fetching code probably won't even be in the drawer state)_

Reply:
That's a fair point. I have removed it from redux. As you mentioned, we can look into storing it somewhere generic when it needs to be reused.